datafusion/core/src/datasource/physical_plan/parquet/mod.rs

```diff
@@ -152,8 +154,9 @@ pub use writer::plan_to_parquet;
 /// the file.
 ///
 /// * Step 3: The `ParquetOpener` gets the [`ParquetMetaData`] (file metadata)
-/// via [`ParquetFileReaderFactory`] and applies any predicates and projections
-/// to determine what pages must be read.
+/// via [`ParquetFileReaderFactory`], creating a [`ParquetAccessPlan`] by
```

Comment:
The main change / reason for this PR is to encapsulate the decision of what row groups / what row selection to scan into a single struct (in this case `ParquetAccessPlan`) so that that the information can be passed in from the outside as well

Aka `ParquetAccessPlan` will be the struct passed in the API described in  https://github.com/apache/datafusion/issues/9929